extensions/oidc/runtime/src/main/java/io/quarkus/oidc/runtime/CodeAuthenticationMechanism.java

```diff
@@ -182,16 +188,19 @@ private static QuarkusSecurityIdentity augmentIdentity(SecurityIdentity security
 
         configContext.auth.authenticate(params, userAsyncResult -> {
             if (userAsyncResult.failed()) {
-                cf.completeExceptionally(new AuthenticationFailedException());
+                if (userAsyncResult.cause() != null) {
+                    LOG.warnf("Exception during the code to token exchange: %s", userAsyncResult.cause().getMessage());
```

Comment:
This should be DEBUG if it is something a user can trigger with a malformed request. In general you don't want to log anything that a user can trigger higher than DEBUG, as it opens you up to DOS attacks because logging is very expensive.

Reply:
@stuartwdouglas done